### PR TITLE
Add support for specific schedule periods in calendars

### PR DESCRIPTION
**Pull request by Kaian**
This PR updates calendar with a new associated entity CalendarPeriods.

This periods can assiociate Schedules that will override External Call filters schedulers within a time period.

### CODE REVIEW

File: library/Ivoz/Provider/Domain/Model/CalendarPeriod/CalendarPeriod.php [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    }
+
+
+    public function isOutOfSchedule()
+    {
+        $calendar = $this->getCalendar();
+        $company = $calendar->getCompany();
+        $timezone = $company->getDefaultTimezone();
+
+        $time = new \DateTime('now', new \DateTimeZone('UTC'));
```
Comment (mmadariaga) [inline library/Ivoz/Provider/Domain/Model/CalendarPeriod/CalendarPeriod.php]: I would call it $now

File: library/Ivoz/Provider/Domain/Model/ExternalCallFilter/ExternalCallFilter.php [hunk 2]
```diff
@@ -153,6 +154,38 @@ public function getHolidayDateForToday()
...
+    public function getCalendarPeriodForToday()
+    {
+        $externalCallFilterRelCalendars = $this->getCalendars();
+        if (empty($externalCallFilterRelCalendars)) {
+            return null;
+        }
+
+        $company = $this->getCompany();
+        $timezone = $company->getDefaultTimezone();
+        $time = new \DateTime('now', new \DateTimeZone($timezone->getTz()));
```
Comment (mmadariaga) [inline library/Ivoz/Provider/Domain/Model/ExternalCallFilter/ExternalCallFilter.php]: $now?

Comment (mmadariaga) [inline library/Ivoz/Provider/Domain/Model/ExternalCallFilter/ExternalCallFilter.php]: $now

File: library/Ivoz/Provider/Infrastructure/Persistence/Doctrine/Mapping/CalendarPeriod.CalendarPeriodAbstract.orm.yml [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    voiceMailUser:
+      targetEntity: \Ivoz\Provider\Domain\Model\User\UserInterface
+      cascade: {  }
+      fetch: LAZY
+      mappedBy: null
+      inversedBy: null
+      joinColumns:
+        voiceMailUserId:
+          referencedColumnName: id
+          onDelete: cascade
```
Comment (cruzccl) [inline library/Ivoz/Provider/Infrastructure/Persistence/Doctrine/Mapping/CalendarPeriod.CalendarPeriodAbstract.orm.yml]: Maybe better ```onDelete: set null``` ?

File: doc/sphinx/administration_portal/client/vpbx/routing_tools/calendars.rst [hunk 4]
```diff
@@ -45,3 +42,30 @@ date with the locution "Happy New Year".
 .. hint:: Holidays without special routing will apply the external call filter
    holiday routing.
 
+
+Calendar Periods
+----------------
+
+Calendars can also be used to override some time periods with a different scheduler.
```
Comment (cruzccl) [inline doc/sphinx/administration_portal/client/vpbx/routing_tools/calendars.rst]: s/scheduler/schedule/

File: library/Ivoz/Provider/Infrastructure/Persistence/Doctrine/Mapping/CalendarPeriod.CalendarPeriodAbstract.orm.yml (hunk 3 above)
Comment (Kaian) [reply library/Ivoz/Provider/Infrastructure/Persistence/Doctrine/Mapping/CalendarPeriod.CalendarPeriodAbstract.orm.yml]: Maybe

**Pull request merged**